cgi/CRMS.pm

```diff
@@ -188,40 +188,33 @@ sub NeedStepUpAuth
   }
   if ($mfa)
   {
-    my ($dbclass, $dbtemplate);
-    $sql = 'SELECT shib_authncontext_class,template FROM ht_institutions'.
+    my $dbclass;
+    $sql = 'SELECT shib_authncontext_class FROM ht_institutions'.
            ' WHERE entityID=? LIMIT 1';
     eval {
       $ref = $sdr_dbh->selectall_arrayref($sql, undef, $idp);
     };
     if ($ref && scalar @{$ref})
     {
-      $dbclass    = $ref->[0]->[0]; # https://refeds.org/profile/mfa or NULL
-      $dbtemplate = $ref->[0]->[1]; # https://___HOST___/Shibboleth.sso/Login?entityID=https://shibboleth.umich.edu/idp/shibboleth&target=___TARGET___
-      #$dbtemplate = 'https://___HOST___/Shibboleth.sso/Login?entityID=___ENTITY_ID___&target=___TARGET___';
+      $dbclass = $ref->[0]->[0]; # https://refeds.org/profile/mfa or NULL
     }
     if (defined $class && defined $dbclass && $class ne $dbclass)
     {
       $need = 1;
-      my $tpl = $dbtemplate;
+      my $template = 'https://___HOST___/Shibboleth.sso/Login?entityID=___ENTITY_ID___&target=___TARGET___';
```

Comment:
I suppose this works, but why not interpolate directly into the template instead of using the regexes?

Reply:
New commit with interpolation (and a couple fixes like URI escaping the target).